File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        
        ListNode delayed = head;
        
        while(head != null && head.next != null && head.next.next != null){
            head = head.next.next;
            delayed = delayed.next;
        }
        
        if(head.next != null){
            delayed =delayed.next;
        }
        
        return delayed;
        
    }
}

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        
        ListNode *delayed = head;
        
        while(head != nullptr && head->next != nullptr && head->next->next != nullptr){
            head = head->next->next;
            delayed = delayed->next;
        }
        
        if(head->next != nullptr) delayed = delayed->next;
        
        return delayed;
        
    }
};